external/s3-sqs/src/main/scala/org/apache/spark/sql/sqs/SqsClient.scala

```diff
@@ -0,0 +1,267 @@
+      implicit val formats = DefaultFormats
+      try {
+        val messageReceiptHandle = message.getReceiptHandle
+        val messageJson = parse(message.getBody).extract[JValue]
+        val bucketName = (
+          messageJson \ "Records" \ "s3" \ "bucket" \ "name").extract[Array[String]].head
+        val eventName = (messageJson \ "Records" \ "eventName").extract[Array[String]].head
+        if (eventName.contains("ObjectCreated")) {
+          val timestamp = (messageJson \ "Records" \ "eventTime").extract[Array[String]].head
+          val timestampMills = convertTimestampToMills(timestamp)
+          val path = "s3://" +
+            bucketName + "/" +
+            (messageJson \ "Records" \ "s3" \ "object" \ "key").extract[Array[String]].head
+          logDebug("Successfully parsed sqs message")
+          list :+ ((path, timestampMills, messageReceiptHandle))
+        } else {
+          if (eventName.contains("ObjectRemoved")) {
+            if (!ignoreFileDeletion) {
+              exception = Some(new SparkException("ObjectDelete message detected in SQS"))
+            } else {
+              logInfo("Ignoring file deletion message since ignoreFileDeletion is true")
+            }
+          } else {
+            logWarning("Ignoring unexpected message detected in SQS")
+          }
+          errorMessages.append(messageReceiptHandle)
+          list
+        }
+      } catch {
+        case me: MappingException =>
+          errorMessages.append(message.getReceiptHandle)
+          logWarning(s"Error in parsing SQS message ${me.getMessage}")
+          list
+        case e: Exception =>
+          errorMessages.append(message.getReceiptHandle)
+          logWarning(s"Unexpected error while parsing SQS message ${e.getMessage}")
+          list
+      }
+    }
+    if (errorMessages.nonEmpty) {
+      addToDeleteMessageQueue(errorMessages.toList)
+    }
+    parsedMessages
+  }
+
+  private def convertTimestampToMills(timestamp: String): Long = {
+    val timeInMillis = timestampFormat.parse(timestamp).getTime()
+    timeInMillis
+  }
+
+  private def evaluateRetries(): Unit = {
+    retriesOnFailure += 1
+    if (retriesOnFailure >= sqsMaxRetries) {
+      logError("Max retries reached")
+      exception = Some(new SparkException("Unable to receive Messages from SQS for " +
+        s"${sqsMaxRetries} times Giving up. Check logs for details."))
+    } else {
+      logWarning(s"Attempt ${retriesOnFailure}." +
+        s"Will reattempt after ${sqsFetchIntervalSeconds} seconds")
+    }
+  }
+
+  private def createSqsClient(): AmazonSQS = {
+    try {
+      val isClusterOnEc2Role = hadoopConf.getBoolean(
+        "fs.s3.isClusterOnEc2Role", false) || hadoopConf.getBoolean(
+        "fs.s3n.isClusterOnEc2Role", false) || sourceOptions.useInstanceProfileCredentials
+      if (!isClusterOnEc2Role) {
+        val accessKey = hadoopConf.getTrimmed("fs.s3n.awsAccessKeyId")
```

Comment:
you should be using the fs.s3a values, given that that the S3n connector has been deleted from hadoop, though I'll add some more comments on that topic later